packages/framework-provider-aws/src/library/read-models-adapter.ts

```diff
@@ -38,14 +45,27 @@ export async function storeReadModel(
   config: BoosterConfig,
   logger: Logger,
   readModelName: string,
-  readModel: ReadModelInterface
+  readModel: ReadModelInterface,
+  expectedCurrentVersion: number
 ): Promise<void> {
-  await db
-    .put({
-      TableName: config.resourceNames.forReadModel(readModelName),
-      Item: readModel,
-    })
-    .promise()
+  try {
+    await db
+      .put({
+        TableName: config.resourceNames.forReadModel(readModelName),
+        Item: readModel,
+        ConditionExpression: 'attribute_not_exists(boosterMetadata.version) OR boosterMetadata.version = :version',
```

Comment:
Interesting. Is this check transactional? I mean, does it exist any possibility that after the check happens and before the write has performed, another thread write succeeds? If it's transactional, this solution is 💯

Reply:
It is guaranteed by DynamoDB to occur atomically 😃 

This is the requirement each provider must fulfil: a way to perform a conditional write atomically